### ISSUE TITLE
Grafana dashboard shows "No Data" for replications with Grafana 8.0.5; works with 7.5.x

### ISSUE THREAD

**Issue by jwittlincohen** [zrepl/zrepl#502]
I recently upgraded Grafana from 7.5.7 to 8.0.5.  Under 7.5.5 and 7.5.7, all my replication jobs show as "No data" whereas with 7.5.5 or 7.5.7, I see "All OK" for the jobs.  The remaining metrics work for both versions.  I tested on a Debian Buster system and I made no other changes to the system.  I'm running zrepl 0.4.0 and Prometheus 2.7.1, neither of which have been updated recently.

8.0.5
![image](https://user-images.githubusercontent.com/9561645/125330380-b0344a00-e314-11eb-886e-c0742ff171d3.png)

7.5.7
![image](https://user-images.githubusercontent.com/9561645/125330276-8b3fd700-e314-11eb-9b81-f48be541a52a.png)

This issue seems very similar to the issue I reported at #363 with Grafana 7.1.4 as it has similar "Data proxy error" syslog messages.  That issue was caused by a regression in Grafana's "repeat-by-variable" feature.

```
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=1 orgId=1 uname=admin path=                                                                        /api/datasources/proxy/1/api/v1/query_range remote_addr=[::1] referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId=1&refresh=5s&var-prom_job_name=promet                                                                        heus&var-zrepl_job_name=All" error="http: proxy error: context canceled"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=1 orgId=1 uname=admin path=                                                                        /api/datasources/proxy/1/api/v1/query_range remote_addr=127.0.0.1 referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId=1&refresh=5s&var-prom_job_name=pr                                                                        ometheus&var-zrepl_job_name=All" error="http: proxy error: context canceled"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET                                                                         path=/api/datasources/proxy/1/api/v1/query_range status=502 remote_addr=[::1] time_ms=88 size=0 referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId=1&r                                                                        efresh=5s&var-prom_job_name=prometheus&var-zrepl_job_name=All"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET                                                                         path=/api/datasources/proxy/1/api/v1/query_range status=502 remote_addr=127.0.0.1 time_ms=85 size=0 referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId                                                                        =1&refresh=5s&var-prom_job_name=prometheus&var-zrepl_job_name=All"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=1 orgId=1 uname=admin path=                                                                        /api/datasources/proxy/1/api/v1/query_range remote_addr=[::1] referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId=1&refresh=5s&var-prom_job_name=promet                                                                        heus&var-zrepl_job_name=All" error="http: proxy error: context canceled"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET                                                                         path=/api/datasources/proxy/1/api/v1/query_range status=502 remote_addr=[::1] time_ms=76 size=0 referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId=1&r                                                                        efresh=5s&var-prom_job_name=prometheus&var-zrepl_job_name=All"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=1 orgId=1 uname=admin path=                                                                        /api/datasources/proxy/1/api/v1/query_range remote_addr=127.0.0.1 referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId=1&refresh=5s&var-prom_job_name=pr                                                                        ometheus&var-zrepl_job_name=All" error="http: proxy error: context canceled"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=1 orgId=1 uname=admin path=                                                                        /api/datasources/proxy/1/api/v1/query_range remote_addr=127.0.0.1 referer=https://cohenltd.net/grafana/ error="http: proxy error: context canceled"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET                                                                         path=/api/datasources/proxy/1/api/v1/query_range status=502 remote_addr=127.0.0.1 time_ms=58 size=0 referer="https://cohenltd.net/grafana/d/etQuvBnGz/zrepl-0-3-new?orgId                                                                        =1&refresh=5s&var-prom_job_name=prometheus&var-zrepl_job_name=All"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET                                                                         path=/api/datasources/proxy/1/api/v1/query_range status=502 remote_addr=127.0.0.1 time_ms=17 size=0 referer=https://cohenltd.net/grafana/
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=1 orgId=1 uname=admin path=                                                                        /api/datasources/proxy/1/api/v1/query_range remote_addr=127.0.0.1 referer=https://cohenltd.net/grafana/ error="http: proxy error: context canceled"
Jul 12 13:18:50 storage-server grafana-server[32745]: t=2021-07-12T13:18:50-0400 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=1 orgId=1 uname=admin path=                                                                        /api/datasources/proxy/1/api/v1/query_range remote_addr=[::1] referer=https://cohenltd.net/grafana/ error="http: proxy error: context canceled"
```

**Comment by problame** [comment]
@jwittlincohen any chance you can dig into this problem yourself? I figured out the problem in #363 by clicking around in the UI...

**Comment by problame** [comment]
Hm, what happens if you do a meaningless edit in the query, save, then reload the dashboard? That's what fixed it the last time IIRC

**Comment by jwittlincohen** [comment]
Nope, no change.  I can confirm that it's still broken as of Grafana 8.2.3 but 7.5.7 continues to work.

**Comment by problame** [comment]
@jwittlincohen I played around with a Gauge instead. No clue what's wrong with the 'Stat' panel type ...

Could you check this one out?
[zrepl 0.3-1636149055143.zip](https://github.com/zrepl/zrepl/files/7489103/zrepl.0.3-1636149055143.zip)

**Comment by jwittlincohen** [comment]
This issue seems resolved as of Grafana 8.2.4.  I installed the latest Grafana package to try out your patched Grafana panel (it works and discovered the existing panel now works properly.  The patched panel works with the gauges but the stat panel shows "No Data".

![old panel](https://user-images.githubusercontent.com/9561645/141820660-58116ce8-0f22-425c-927f-601dc4ba1cd9.PNG)

![patched panel](https://user-images.githubusercontent.com/9561645/141820655-85cb1eaa-9ce6-4579-899c-f429c7450710.PNG)

**Comment by problame** [comment]
Let's stick with the old panel then.
